cedar-policy-generators/src/hierarchy.rs

```diff
@@ -95,17 +95,34 @@ impl Hierarchy {
         }
     }
 
-    /// generate an arbitrary uid based on the hierarchy
-    pub fn arbitrary_uid(&self, u: &mut Unstructured<'_>) -> Result<EntityUID> {
+    /// Generate an arbitrary uid based on the hierarchy. If `request_field`
+    /// is `Some(var)` then the generated uid may be unspecified with eid `var`.
+    /// Otherwise, the uid is guaranteed to be specified.
+    pub fn arbitrary_uid(
+        &self,
+        u: &mut Unstructured<'_>,
+        request_field: Option<ast::Var>,
+    ) -> Result<EntityUID> {
         // UID that exists or doesn't. 90% of the time pick one that exists
         if u.ratio::<u8>(9, 10)? {
             let uid = u
                 .choose(&self.uids)
                 .map_err(|e| while_doing("getting an arbitrary uid".into(), e))?;
             Ok(uid.clone())
         } else {
-            // Note: may generate an unspecified entity
```

Comment:
Did we want to drop this note?

Reply:
I think this is covered by other comments added.